engine/datahistory_manager_test.go

```diff
@@ -1530,10 +1533,13 @@ func dataHistoryTraderLoader(exch, a, base, quote string, start, _ time.Time) ([
 	}, nil
 }
 
-func dataHistoryCandleLoader(exch string, cp currency.Pair, a asset.Item, i kline.Interval, start, _ time.Time) (*kline.Item, error) {
+func dataHistoryCandleLoader(exch string, cp currency.Pair, a asset.Item, i kline.Interval, start, end time.Time) (*kline.Item, error) {
 	start = start.Truncate(i.Duration())
 	var candles []kline.Candle
-	for x := 0; x < 24; x++ {
+	for {
+		if start.Equal(end) || start.After(end) {
+			break
```

Comment:
might return an error? These seem funky?

Reply:
The list of 24 didn't work with certain start and end dates. this just populates the appropriate amount of data required. Would you like me to add some more checks?